e2e/inspect/inspect.go

```diff
@@ -0,0 +1,182 @@
+
+const expectedTestJson = `
+{
+	"attributes": {
+		"apps": "",
+		"test": "#!/bin/sh\n\nls /\ntest -d /\ntest -d /etc\n\n\n"
+	},
+	"type": "container"
+}`
+
+const expectedEnvironmentJson = `
+{
+	"attributes": {
+		"apps": "",
+		"environment": {
+			"90-environment.sh": "#!/bin/sh\n#Custom environment shell code should follow\n\n\nexport test=\"testing\"\nexport e2e=\"e2e testing\"\n\n\n"
+		}
+	},
+	"type": "container"
+}`
+
+func runInspectCommand(inspectType string) ([]byte, error) {
+	argv := []string{"inspect", "--json", inspectType, "testdata/test.sif"}
+	cmd := exec.Command("singularity", argv...)
+
+	return cmd.CombinedOutput()
+}
+
+func singularityInspect(t *testing.T) {
+	tests := []struct {
+		name      string
+		insType   string
+		json      []string
+		expectOut string
+	}{
+		{
+			name:      "label E2E",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "E2E"},
+			expectOut: expectedLabelsJson,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "HI"},
+			expectOut: expectedLabelsJson,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "e2e"},
+			expectOut: expectedLabelsJson,
+		},
+		{
+			name:      "label",
+			insType:   "--labels",
+			json:      []string{"attributes", "labels", "hi"},
+			expectOut: expectedLabelsJson,
+		},
+		{
+			name:      "runscript",
+			insType:   "--runscript",
+			json:      []string{"attributes", "runscript"},
+			expectOut: expectedRunscriptJson,
+		},
+		{
+			name:      "list apps",
+			insType:   "--list-apps",
+			json:      []string{"attributes", "apps"},
+			expectOut: expectedListAppsJson,
+		},
+		{
+			name:      "test",
+			insType:   "--test",
+			json:      []string{"attributes", "test"},
+			expectOut: expectedTestJson,
+		},
+		{
+			name:      "environment",
+			insType:   "--environment",
+			json:      []string{"attributes", "environment", "90-environment.sh"},
+			expectOut: expectedEnvironmentJson,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, test.WithPrivilege(func(t *testing.T) {
+			out, err := runInspectCommand(tt.insType)
+			if err != nil {
+				t.Fatalf("Unexpected failure: %s: %s", string(out), err)
+			}
+
+			// Check the E2E label in test.sif, does it match our expected output
+			v, err := jsonparser.GetString(out, tt.json...)
+			if err != nil {
+				fmt.Println("ERROR: ", err)
+			}
+			// Get the expected output, and compair them
+			e, err := jsonparser.GetString([]byte(tt.expectOut), tt.json...)
+			if err != nil {
+				t.Fatalf("Unable to get expected output from json: %v", err)
+			}
+			if v != e {
+				t.Fatalf("Unexpected faulure: got: %s, expecting: %s", v, e)
+			}
+
+		}))
+	}
+
+}
+
+// RunE2ETests is the main func to trigger the test suite
+func RunE2ETests(t *testing.T) {
+	e2e.LoadEnv(t, &testenv)
+
+	// try to build from a non existen path
+	t.Run("singularityEnv", singularityInspect)
```

Comment:
"singularityEnv" seems a little too generic.

Reply:
Also copy-paste error, should be 

```go
t.Run("singularityInspect", singularityInspect)
```